test/blackbox-tests/test-cases/runtest-cmd.t

```diff
@@ -0,0 +1,59 @@
+Here we test the features of the `dune runtest` command. 
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.16)
+  > EOF
+
+  $ cat > mytest.t <<EOF
+  >   $ echo "Hello, world!"
+  > "Goodbye, world!"
+  > EOF
+  $ mkdir -p tests/myothertest.t
+  $ cat > tests/myothertest.t/run.t <<EOF
+  >   $ echo "Hello, world!"
+  > "Goodbye, world!"
+  > EOF
+  $ cat > tests/filetest.t <<EOF
+  >   $ echo "Hello, world!"
+  > "Goodbye, world!"
+  > EOF
+
+Passing no arguments to `dune runtest` should be equivalent to `dune build
+@runtest`.
+
+  $ dune test 2>&1 | grep "^File"
+  File "mytest.t", line 1, characters 0-0:
+  File "tests/filetest.t", line 1, characters 0-0:
+  File "tests/myothertest.t/run.t", line 1, characters 0-0:
+
+Passing the name of a test should only run that test.
+
+  $ dune test mytest 2>&1 | grep "^File"
```

Comment:
I'd say we should just accept the full filename. That will work better with tab completion.

Reply:
The new commit makes only the valid path names work.